src/bicgstab.cpp

```diff
@@ -96,11 +96,11 @@ static double norm2(int n, const realnum *x) {
     double xs = scale * x[i];
     sum += xs * xs;
   }
-  return xmax * sqrt(sum_to_all(sum));
+  return static_cast<double>(xmax * sqrt(sum_to_all(sum)));
```

Comment:
The type of sum is long double. Implicit conversion to double upon return will trigger a compiler warning without static cast.